src/mscorlib/shared/System/Collections/Generic/Dictionary.cs

```diff
@@ -1041,6 +1191,9 @@ ICollection IDictionary.Values
                 {
                     ThrowHelper.ThrowWrongKeyTypeArgumentException(key, typeof(TKey));
                 }
+#if DEBUG
```

Comment:
❓ Why bother with `#if DEBUG` here if the target method is marked `[Conditional("DEBUG")]`?

Reply:
If you do that it might be worth naming something like as DebugOnlyConcurrentAccessCheck so it's obvious it's debug only

Reply:
Wasn't sure the Jit would elide all the `int version = _version;` calls (or C# delete them and not produce an error about not using it)

Reply:
Will change an check diffs as it reduces the number of `#if`s considerably